docs/proving-contract-properties.md

```diff
@@ -0,0 +1,26 @@
+# Goal:
+Verify properties of a contract starting from simple (Crowdfunding) to more complex (ICO).
+
+# Workflow:
+Write the contract in SigmaDls.
+During the compilation:
+* Our macros generate ErgoTree;
+* [Stainless](https://github.com/epfl-lara/stainless) verifies contract properties;
+
+# Plan:
+
+## Crowdfunding Contract:
+- [x] some properties of the Crowdfunding contract are already proven, see [source](http://github.com/ScorexFoundation/sigmastate-interpreter/blob/1f65d90cdab0388acc73bfa43797e39d3d9e32bc/contract-verification/src/main/scala/sigmastate/verification/contract/CrowdFundingContractVerification.scala#L34-L76)
```

Comment:
This is really nice. I wonder if we can formulate and verify mutual exclusion of contract preconditions.

Reply:
There is only one `require` allowed. So all preconditions should be together.
I'm not sure I get the idea of what do you want to verify here. I've used preconditions for:
1. to set a list of requirements for the input in the contract itself;
2. to "limit" the inputs in proofs (see `prove*` methods);